modules_basic/avatar/profile.js

```diff
@@ -9,71 +16,71 @@ exports.needs = {
   followers: 'first'
 }
 
-exports.gives = 'avatar_profile'
-
-function streamToList(stream, el) {
-  pull(
-    stream,
-    pull.drain(function (item) {
-      if(item) el.appendChild(item)
-    })
-  )
-  return el
+exports.gives = {
+  avatar_profile: true,
+  mcss: true
 }
 
 exports.create = function (api) {
-
-  function image_link (id) {
-    return api.avatar_image_link(id, 'thumbnail')
+  return { 
+    avatar_profile,
+    mcss: () => fs.readFileSync(__filename.replace(/js$/, 'mcss'), 'utf8')
   }
 
-  return function (id) {
-
-    var follows_el = h('div.profile__follows.wrap')
-    var friends_el = h('div.profile__friendss.wrap')
-    var followers_el = h('div.profile__followers.wrap')
-    var a, b
-
-    pull(api.follows(id), pull.unique(), pull.collect(function (err, ary) {
-      a = ary || []; next()
-    }))
-    pull(api.followers(id), pull.unique(), pull.collect(function (err, ary) {
-      b = ary || {}; next()
-    }))
-
-    function next () {
-      if(!(a && b)) return
-      var _c = [], _a = [], _b = []
+  function avatar_profile (id) {
 
-      a.forEach(function (id) {
-        if(!~b.indexOf(id)) _a.push(id)
-        else               _c.push(id)
-      })
-      b.forEach(function (id) {
-        if(!~_c.indexOf(id)) _b.push(id)
-      })
-      function add (ary, el) {
-        ary.forEach(function (id) { el.appendChild(image_link(id)) })
-      }
-
-      add(_a, follows_el)
-      add(_c, friends_el)
-      add(_b, followers_el)
-    }
+    var rawFollows = MutantArray()
+    var rawFollowers = MutantArray()
+    var friends = computed([rawFollows, rawFollowers], (follows, followers) => {
+      return follows.filter(follow => followers.includes(follow))
+    })
 
+    var follows = computed([rawFollows, friends], (follows, friends) => {
+      return follows.filter(follow => !friends.includes(follow))
+    })
+    var followers = computed([rawFollowers, friends], (followers, friends) => {
+      return followers.filter(follower => !friends.includes(follower))
+    })
 
-    return h('div.column.profile',
-      api.avatar_edit(id),
-      api.avatar_action(id),
-      h('div.profile__relationships.column',
-        h('strong', 'follows'),
-        follows_el,
-        h('strong', 'friends'),
-        friends_el,
-        h('strong', 'followers'),
-        followers_el
+    pull(
+      api.follows(id),
+      unique(), 
+      drain(
+        peer => rawFollows.push(peer), 
+        data => console.log('follows drain done', data)
       )
     )
+    pull(
+      api.followers(id),
+      unique(), 
+      drain(
+        peer => rawFollowers.push(peer), 
+        data => console.log('followers drain done', data)
```

Comment:
yeah sorry that was a vauge message - was running to get out the door. 

What I'm seeing is this logged  :
```js
{message: "could not find matching timestamp for index:{"valu…25519","sequence":246},"timestamp":1481442875351}", name: "Error", stack: "Error: could not find matching timestamp for index…ttlebot/node_modules/level-sublevel/nut.js:121:19"}
```

I think it's a preature abort of the pull stream (I've read `pull.drain` a bit but some of the switches  I don't quite follow). 

I've looked closely at friendships and followers , and the 'status'  and think I might've seen it say "you are friends" but not seen myself in the friendship list.

Probably useful to understand what pull.drain is doing, and how best to handle that error - without the callback, it's breaking the interface....